app/models/concerns/translatable.rb

```diff
@@ -74,13 +120,73 @@ def set_translations(attribute, translations = {})
     end
   end
 
-  # returns the element from the parent objects assocation list for the given object's id
+
+  # TODO: this might not be the apropritate location for this helper
+  # Returns a hash of the translated terms in all permitted locales for this translatable
+  def translate(*terms)
+    permitted_locales.each_with_object({}) {|l, res|
+      res[l] = terms.each_with_object({}) {|term, res|
+        res[term] = I18n.t(term, locale: l)
+      }}
```

Comment:
I assume this is not consulting the database translations? If so, I agree it doesn't belong here if the only thing it is consulting from the model is the permitted_locales attribute.

Also please use `do..end` for multiline blocks.